src/analysis/transcendental/liouville.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import data.polynomial.denoms_clearable
+import ring_theory.algebraic
+import topology.algebra.polynomial
+import analysis.calculus.mean_value
+import data.real.liouville
+/-!
+# Liouville's theorem
+
+This file contains the proof of Liouville's theorem stating that all Liouville numbers are
+transcendental.
+-/
+
+open set ring_hom real
+
+section inequality_and_intervals
+
+lemma mem_Icc_iff_abs_le {R : Type*} [linear_ordered_add_comm_group R] {x y z : R} :
+  abs (x - y) ≤ z ↔ y ∈ Icc (x - z) (x + z) :=
+⟨λ h, ⟨sub_le.mp (abs_le.mp h).2, neg_le_sub_iff_le_add.mp (abs_le.mp h).1⟩,
+ λ hy, abs_le.mpr ⟨neg_le_sub_iff_le_add.mpr hy.2, sub_le.mp hy.1⟩⟩
+
+end inequality_and_intervals
+
+--namespace is_liouville
```

Comment:
Remove?

Reply:
Dear Scott,

I had forgotten about the `help-wanted` label.  I am merging this PR in small bits and I believe that all that is left from this one is the file `liouville_constant`.  I am going to leave this PR here as a blueprint, but will remove it, once all of it is merged.

Does this seem reasonable?

Reply:
Actually, this is a different PR than the one that I thought.  Still, it seems that this one will get absorbed by bits in what I am pushing these days.